### ISSUE TITLE
Got JS Exception: undefined is not an object (evaluating 't.remoteModuleConfig')

### ISSUE THREAD

**Issue by JimmyDaddy** [facebook/react-native#15785]
### Is this a bug report?

Yes

### Environment

1. `react-native -v`: react-native-cli: 2.0.1    react-native: 0.47.1
2. `node -v`:v8.4.0
3. `npm -v`:5.4.0

Then, specify:

- Target Platform:
IOS
- Development Operating System:
MacOS
- Build tools:
Xcode
### Steps to Reproduce

1.use Xcode archive a release IPA
2.install the IPA in an iPhone
3.open the app, it starts successfully at once, then get white screen and crash
4. reopen the APP, it will crash

### error message
 ```
2017-09-04 20:16:25.383975+0800 natasha[3621:644663] [] nw_connection_get_connected_socket_block_invoke 13 Connection has no connected handler
2017-09-04 20:16:28.404 [error][tid:com.facebook.react.JavaScript] undefined is not an object (evaluating 't.remoteModuleConfig')
2017-09-04 20:16:28.404994+0800 natasha[3621:644853] undefined is not an object (evaluating 't.remoteModuleConfig')
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0904 20:16:28.443285 1845309440 JSCHelpers.cpp:104] Got JS Exception: undefined is not an object (evaluating 't.remoteModuleConfig') (/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:85)
E0904 20:16:28.443593 1845309440 JSCHelpers.cpp:110] Got JS Stack: /var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:85:1100
get@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:85:366
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:83:101
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:82:53
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:102:51
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:245:225
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:205:915
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:204:56
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:68:3464
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:12:109
o@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:505
e@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:2:244
global code@/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:858:9
2017-09-04 20:16:28.447 [fatal][tid:main] Unhandled JS Exception: undefined is not an object (evaluating 't.remoteModuleConfig') (/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:85)
2017-09-04 20:16:28.446767+0800 natasha[3621:644538] Unhandled JS Exception: undefined is not an object (evaluating 't.remoteModuleConfig') (/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:85)
2017-09-04 20:16:28.448351+0800 natasha[3621:644538] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: undefined is not an object (evaluating 't.remoteModuleConfig') (/var/mobile/Containers/Data/Application/A9EBB51C-07A1-4B96-A8AB-E5490A0F7F4C/Library/JSCode/doctor.jsbundle:85)', reason: 'Unhandled JS Exception: undefined is not an object (evaluating 't.remoteMod...'
*** First throw call stack:
(0x18eba2fe0 0x18d604538 0x18eba2f28 0x100130a80 0x100118c9c 0x100ca9a50 0x100ca9a10 0x100caeb78 0x18eb510c8 0x18eb4ece4 0x18ea7eda4 0x1904e9074 0x194d39c9c 0x10001a458 0x18da8d59c)
libc++abi.dylib: terminating with uncaught exception of type NSException
warning: could not execute support code to read Objective-C class data in the process. This may reduce the quality of type information available.
Printing description of name->isa:
(Class) isa = 0x000001a1b4e80d51
(lldb) 
```

**Comment by koansang** [comment]
+ 1 same issue..

**Comment by esauter5** [comment]
Have you recently upgraded CodePush? I ran into this from that. I'm guessing it had something to do with the fact that I hadn't updated my app version so it was actually pulling a version of my app that used the old CodePush and I'm assuming some API change broke that.

tldr - make sure youre not downloading an update of your app that uses an older code push version. I bumped my android version of my app and everything went fine.

**Comment by esauter5** [comment]
Scratch that. I thought I was in the CodePush repo haha. Regardless, my fix still worked :shrug:

**Comment by JimmyDaddy** [comment]
@esauter5 I'm using my own hot-update module.  There may be something wrong with checking update. When the server returns a right update info(which is a JSON data) it will be good, if the server returns a bad content it will be broken. I know what causes this problem, but don't know why

**Comment by raff90** [comment]
having same issue
